pwndbg/inthook.py

```diff
@@ -12,26 +12,32 @@
 
 import os
 import sys
+import enum
 
 import gdb
 from future.utils import with_metaclass
 
 import pwndbg.typeinfo
 
-if sys.version_info < (3,0):
+if sys.version_info < (3, 0):
     import __builtin__ as builtins
 else:
     import builtins
 
 _int = builtins.int
 
+
 # We need this class to get isinstance(7, xint) to return True
 class IsAnInt(type):
     def __instancecheck__(self, other):
         return isinstance(other, _int)
 
+
 class xint(with_metaclass(IsAnInt, builtins.int)):
     def __new__(cls, value, *a, **kw):
+        if isinstance(cls, enum.EnumMeta):
```

Comment:
It might actually be better to have this at the end of the two `if` statements, and change them to `if` / `elif` so that we catch all unknown types.